internal/sync/sync.go

```diff
@@ -113,15 +112,16 @@ func (s *Synchronizer) setStateDiffCollector(apiSync bool) {
 
 // Run starts the service.
 func (s *Synchronizer) Run() {
-	s.Running = true
 	go s.updateBlocksInfo()
 	go s.handleSync()
 }
 
 func (s *Synchronizer) handleSync() {
 	s.wg.Add(1)
 	for {
+		s.Running = true
 		if err := s.sync(); err != nil {
+			s.Running = false
 			s.logger.With("Error", err).Info("Sync Failed, restarting iterator in 10 seconds")
 			time.Sleep(10 * time.Second)
 			s.stateDiffCollector.Close()
```

Comment:
Can the logic on lines 122, 124, and 127 be moved inside the `s.sync()` method? The reason is that looks like is the responsibility of that method to manage that and not of the caller, and then the `s.handleSync()` method only just cares about the retry

Reply:
Done using defer, let me know wdyt